### ISSUE TITLE
The loss value between the brackets in the logging of training

### ISSUE THREAD

**Issue by a-akram-98** [facebookresearch/detr#333]
Hello, there is something in the logging of the training I can't relate, Every loss has a number and another number between the brackets whats is the number between the brackets and where is line that is responsible for printing this number between the brackets

**Comment by aedirn** [comment]
I am wondering the same thing, especially since sometimes, the loss value fluctuates wildly while the loss value between parentheses/brackets fluctuates very little. 

I found where the loss and other metrics gets formatted/printed, in def __str__(self) in misc.py around line 178. It only appears to print the key: value pair, not the parenthesized value after it. Repo authors, it would be lovely to understand what the parenthesized value is telling us :)

**Comment by aedirn** [comment]
Hi admins, 

These parenthesized values have become more important for me. I implemented a simple check to save the model version with the best test loss. Accessing `test_stats['loss']` gives the parenthesized value (in this case, 8.2174). Where does this value come from? Why does it differ from the un-parenthesized value (in this case, 6.6880)? 

Here is a copy-paste of part of my output from one of the epochs: 
`
Averaged stats: class_error: 90.00  loss: 6.6880 (8.2174)  loss_bbox: 0.0820 (0.1172)  loss_bbox_0: 0.0830 (0.1157)  loss_bbox_1: 0.0872 (0.1171)  loss_bbox_2: 0.0824 (0.1148)  loss_bbox_3: 0.0812 (0.1167)  loss_bbox_4: 0.0851 (0.1177)  loss_ce: 0.5227 (0.4922)  loss_ce_0: 0.5067 (0.4880)  loss_ce_1: 0.4636 (0.4870)  loss_ce_2: 0.4212 (0.4730)  loss_ce_3: 0.4359 (0.4799)  loss_ce_4: 0.5139 (0.5037)  loss_giou: 0.7055 (0.7887)  loss_giou_0: 0.7131 (0.7569)  loss_giou_1: 0.6386 (0.7609)  loss_giou_2: 0.6696 (0.7639)  loss_giou_3: 0.6356 (0.7725)  loss_giou_4: 0.6265 (0.7516)  cardinality_error_unscaled: 1.1250 (1.3281)  cardinality_error_0_unscaled: 1.6250 (2.4609)  cardinality_error_1_unscaled: 1.6250 (2.2109)  cardinality_error_2_unscaled: 1.1250 (1.5547)  cardinality_error_3_unscaled: 1.1250 (1.4375)  cardinality_error_4_unscaled: 1.0000 (1.2578)  class_error_unscaled: 62.5000 (72.1658)  loss_bbox_unscaled: 0.0164 (0.0234)  loss_bbox_0_unscaled: 0.0166 (0.0231)  loss_bbox_1_unscaled: 0.0174 (0.0234)  loss_bbox_2_unscaled: 0.0165 (0.0230)  loss_bbox_3_unscaled: 0.0162 (0.0233)  loss_bbox_4_unscaled: 0.0170 (0.0235)  loss_ce_unscaled: 0.5227 (0.4922)  loss_ce_0_unscaled: 0.5067 (0.4880)  loss_ce_1_unscaled: 0.4636 (0.4870)  loss_ce_2_unscaled: 0.4212 (0.4730)  loss_ce_3_unscaled: 0.4359 (0.4799)  loss_ce_4_unscaled: 0.5139 (0.5037)  loss_giou_unscaled: 0.3527 (0.3944)  loss_giou_0_unscaled: 0.3565 (0.3784)  loss_giou_1_unscaled: 0.3193 (0.3804)  loss_giou_2_unscaled: 0.3348 (0.3820)  loss_giou_3_unscaled: 0.3178 (0.3862)  loss_giou_4_unscaled: 0.3133 (0.3758)
`

You can see all of the values followed by parenthesized values. Why are they both displayed, and what do they each mean?

**Comment by a-akram-98** [comment]
@aedirn the assertion `assert (boxes2[:, 2:] >= boxes2[:, :2]).all()` in the giou function in box_ops.py, gives an error and I checked my dataset and `boxes2[:, 2:] >= boxes2[:, :2]` is true but in the function it self it gives me false

**Comment by aedirn** [comment]
> @aedirn the assertion `assert (boxes2[:, 2:] >= boxes2[:, :2]).all()` in the giou function in box_ops.py, gives an error and I checked my dataset and `boxes2[:, 2:] >= boxes2[:, :2]` is true but in the function it self it gives me false

That's off-topic to this issue, but there are other issues dedicated to the same error: 
https://github.com/facebookresearch/detr/issues/232 
https://github.com/facebookresearch/detr/issues/101 
https://github.com/facebookresearch/detr/issues/28
Are you using custom dataloader code, or any format other than MSCOCO?

**Comment by mnc537** [comment]
The pair of values for each metric corresponds to its median and global average over a window of size 20 (default) and it is defined [here](https://github.com/facebookresearch/detr/blob/master/util/misc.py#L33). They are initialized in line 160 (misc.py) as SmoothedValue.

**Comment by aedirn** [comment]
> The pair of values for each metric corresponds to its median and global average over a window of size 20 (default) and it is defined [here](https://github.com/facebookresearch/detr/blob/master/util/misc.py#L33). They are initialized in line 160 (misc.py) as SmoothedValue.

Thank you so much! This answers it perfectly. Totally missed that when reading the code.

**Comment by a-akram-98** [comment]
Thank you both very much you really helped me
